openedx/adg/admin_dashboard/applications/helpers.py

```diff
@@ -0,0 +1,95 @@
+    submitted applications that have been applied against the business unit that the user is admin of.
+
+    Arguments:
+        user (User): admin for whom submitted applications are to be fetched
+
+    Returns:
+        QuerySet: queryset containing submitted applications
+    """
+    submitted_applications = UserApplication.objects.prefetch_related(
+        'user__application_hub__submission_date').filter(
+        user__application_hub__is_written_application_completed=True)
+
+    if not (user.is_superuser or is_adg_admin(user)):
+        submitted_applications = submitted_applications.filter(business_line__group=user.groups.first())
+
+    return submitted_applications
+
+
+def get_new_applications_and_percentage_change(submitted_applications):
+    """
+    Calculates the number of new applications in the current month and the percentage change between the number of
+    submitted applications in the current and previous month.
+
+    Arguments:
+        submitted_applications (QuerySet): queryset containing submitted applications
+
+    Returns:
+         new_applications (int): total number of submitted application in the current month
+         percentage (int): percentage change between submitted applications in the current and previous month
+    """
+    today = datetime.today()
+    start_of_month = datetime(today.year, today.month, 1)
+    start_of_previous_month = datetime(today.year, today.month - 1, 1)
+
+    new_applications = submitted_applications.filter(
+        user__application_hub__submission_date__range=[start_of_month, today]).count()
+
+    previous_month_applications = submitted_applications.filter(
+        user__application_hub__submission_date__range=[start_of_previous_month, start_of_month - timedelta(days=1)]
+    ).count()
+
+    try:
+        percentage = int((new_applications - previous_month_applications) / previous_month_applications * 100)
+    except ZeroDivisionError:
+        percentage = new_applications * 100
+
+    return new_applications, percentage
+
+
+def get_weekly_application_submissions(submitted_applications):
+    """
+    Calculates the number of submitted applications for the past 12 weeks including the current week.
+
+    Arguments:
+        submitted_applications (QuerySet): queryset containing submitted applications
+
+    Returns:
+        JSON: a json object containing two lists: one for dates of the past 12 weeks and the other for number of
+        application submissions in those weeks
+    """
+    today = datetime.today()
+    start_of_week = today - timedelta(days=today.weekday())
+
+    submissions_list = []
+    date_list = []
+    start_date = start_of_week
+    end_date = today
+
+    for _ in range(1, 13):
+        applications = submitted_applications.filter(
+            user__application_hub__submission_date__range=[start_date, end_date]).count()
+
+        submissions_list.insert(0, applications)
```

Comment:
since insert is an O(1) operation, for better time complexity we can loop from `for i in range(12, -1, -1):`
and insert like submission_list[i] = applications, resulting in O(1) insertions.

But if you think code would be uncleaner in my suggested approach, let's keep things as it is because here we're talking about only 12 insertions, and it won't be a heavy operation practically speaking.